Library/ViewModels/DiscoveryFiltersViewModel.swift

```diff
@@ -59,28 +65,55 @@ public final class DiscoveryFiltersViewModel: DiscoveryFiltersViewModelType,
       .take(first: 1)
       .map { topFilters(forUser: AppEnvironment.current.currentUser) }
 
-    let initialSelectedRowWithCategories = Signal.combineLatest(
-      self.initialSelectedRowWithCategoriesProperty.signal.skipNil(),
+    let initialSelectedRow = Signal.combineLatest(
+      self.initialSelectedRowProperty.signal.skipNil(),
       self.viewDidLoadProperty.signal
       )
       .take(first: 1)
       .map(first)
 
     let topRows = Signal.combineLatest(
       initialTopFilters,
-      initialSelectedRowWithCategories.map(first)
+      initialSelectedRow
       )
       .map { params, selectedRow -> [SelectableRow] in
         params.map { p in SelectableRow(isSelected: p == selectedRow.params, params: p) }
     }
 
-    let categoryId = self.initialSelectedRowWithCategoriesProperty.signal.skipNil()
-      .map(first)
+    let categoryId = self.initialSelectedRowProperty.signal.skipNil()
       .map { $0.params.category?.rootId }
 
+    let areLoading = MutableProperty(false)
+    self.categoriesAreLoading = areLoading.signal
+
+    self.yConstantForActivityIndicator = self.categoriesAreLoading
+      .map { _ in AppEnvironment.current.currentUser == nil ? -Styles.grid(15) : Styles.grid(10) }
+      .take(first: 1)
+
+    let cachedCats = self.viewDidLoadProperty.signal
+      .map(cachedCategories)
+
+    let categoriesEvent = cachedCats
+      .filter { $0 == nil || ($0?.isEmpty ?? true) }
```

Comment:
a more direct way to express this is `$0?.isEmpty != .some(false)`, which evaluates to `true` for both `nil` and `true`.